### ISSUE TITLE
[QUERY] EventHubConsumerClient - exceeded the maximum number of allowed receivers per partition in a consumer group which is 5, for one single read thread

### ISSUE THREAD

**Issue by anlong465** [Azure/azure-sdk-for-net#40836]
### Library name and version

Azure.Messaging.EventHubs.Consumer 5.15.5

### Query/Question

I was headache by this exception for a couple of months: "Azure.Messaging.EventHubs.EventHubsException(QuotaExceeded): Exceeded the maximum number of allowed receivers per partition in a consumer group which is 5."  

I did some google, and found https://github.com/Azure/azure-sdk-for-net/issues/33938  But my scenario is a little different, and then I have to create a new one: 

Anyway, a brief of my scenario is:
- every partition has one specific k8s/pod to process its own message
- inside such pod, only one long-running thread as well as one cached EventHubConsumerClient object keeps fetching message from such partition: 
- and then after a couple of weeks or days, one exception similar as bellow would block my pod: after such exception there is not any new log message printout, but k8s health probe still think it is healthy.
Exception in thread "main" com.azure.core.amqp.exception.AmqpException: Exceeded the maximum number of allowed receivers per partition in a consumer group which is 5. **List of connected receivers - f3cf139f-ea40-46e0-be7e-7499efc062b4, f3cf139f-ea40-46e0-be7e-7499efc062b4, f3cf139f-ea40-46e0-be7e-7499efc062b4, f3cf139f-ea40-46e0-be7e-7499efc062b4, f3cf139f-ea40-46e0-be7e-7499efc062b4.**

The weird thing in the above exception message is: 1) all listed receivers share one same UUID; 2) the related exception stack trace doesn't include any lines of our own codes, and then we don't know how to catch such exception

- Below here it is fetching message logic: inside one for-ever while(true) loop: for every iteration, it tries to fetch out at most 100 messages in 1 second from such partition and then do the corresponding logic; if the total message counter for such iteration is less than 50, the current thread would sleep 500 milliseconds; and then repeats for the next iteration.

**Comment by jsquire** [comment]
Hi @anlong465.   Thank you for reaching out and we regret that you're experiencing difficulties.  The error message that you're seeing indicates that you have more than 5 different consumer types attempting to read from a partition using the same consumer group.   

There's a few things here that we'll need to figure out before we can assist:

- What package and version are you using?   There is no `Azure.Messaging.EventHubs.Consumer` package, nor do the  .NET `Azure.Messaging.EventHubs` packages have a version 5.15.5.
- What language are you using?  The package version and AMQP error would indicate Java, but your stated error type would indicate .NET.
- Please provide the full error details, including the error message and stack trace.
- Please provide an example of how you're using the client - particularly how you're initializing it and managing its lifetime.

**Comment by anlong465** [comment]
Hi @jsquire , thanks a lot for your timely resposne and sorry for the mess.
1) **I am using Java. and the related maven dependency is:**
         <dependency>
            <groupId>com.azure</groupId>
            <artifactId>azure-messaging-eventhubs</artifactId>
            <version>5.15.5</version>
        </dependency>

2) **Regarding with full stack trace:** all log4j errors are stored into database, via the database record, I have the below finding:
_com.azure.core.amqp.exception.AmqpException: Exceeded the maximum number of allowed receivers per partition in a consumer group which is 5. List of connected receivers - 5fed4790-f44b-4054-a44a-5942dcb616c3, 5fed4790-f44b-4054-a44a-5942dcb616c3, 5fed4790-f44b-4054-a44a-5942dcb616c3, 5fed4790-f44b-4054-a44a-5942dcb616c3, 5fed4790-f44b-4054-a44a-5942dcb616c3. TrackingId:d3ec969b-c2be-42ac-b064-1bdb23227f4d_B42, SystemTracker:brewdat-supply-eventhub-prod:eventhub:brewdat-supply-prod~16383, Timestamp:2023-12-08T06:14:40 Reference:856b75db-575b-41ff-be39-682e4ba602de, TrackingId:09c995eb-a30c-4db8-98a7-e2b5db3a711e_B42, SystemTracker:brewdat-supply-eventhub-prod:eventhub:brewdat-supply-prod~16383|brewdat-iot-monitor, Timestamp:2023-12-08T06:14:40 TrackingId:0f08adfb0dd5407eb9292108628952c7_G29, SystemTracker:gateway5, Timestamp:2023-12-08T06:14:40, errorContext[NAMESPACE: brewdat-supply-eventhub-prod.servicebus.chinacloudapi.cn. ERROR CONTEXT: N/A, PATH: brewdat-supply-prod/ConsumerGroups/brewdat-iot-monitor/Partitions/0, REFERENCE_ID: 0_aa5e3c_1702016080539, LINK_CREDIT: 0]
	at com.azure.core.amqp.implementation.ExceptionUtil.toException(ExceptionUtil.java:85)
	at com.azure.core.amqp.implementation.handler.LinkHandler.handleRemoteLinkClosed(LinkHandler.java:88)
	at com.azure.core.amqp.implementation.handler.LinkHandler.onLinkRemoteClose(LinkHandler.java:38)
	at com.azure.core.amqp.implementation.handler.ReceiveLinkHandler.onLinkRemoteClose(ReceiveLinkHandler.java:228)
	at org.apache.qpid.proton.engine.BaseHandler.handle(BaseHandler.java:176)
	at org.apache.qpid.proton.engine.impl.EventImpl.dispatch(EventImpl.java:108)
	at org.apache.qpid.proton.reactor.impl.ReactorImpl.dispatch(ReactorImpl.java:324)
	at org.apache.qpid.proton.reactor.impl.ReactorImpl.process(ReactorImpl.java:291)
	at com.azure.core.amqp.implementation.ReactorExecutor.run(ReactorExecutor.java:91)
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:68)
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:28)
	at java.ut_
![1702953164186](https://github.com/Azure/azure-sdk-for-net/assets/53285941/69d3350e-b843-4401-b074-5f35b5f3b6e6)

3) Regarding with initializing
    _protected EventHubConsumerClient makeEventHubConsumerClient() {
        return new EventHubClientBuilder()
                .connectionString(connectionString)
                .consumerGroup(consumerGroup)
                .buildConsumerClient();
    }
    private final EventHubConsumerClient consumer = makeEventHubConsumerClient() ;
   _

4) Regarding with managing its lifetime
Actually, I never close it as my thread is one long-running thread, and besides of fetching out message,  I didn't do any management against it.
Below here it is my Thread.run() method, 
![1702954274268](https://github.com/Azure/azure-sdk-for-net/assets/53285941/e3c92821-c60b-4411-93f5-2b63250f09e8)

And here it is to fetch out message from EventHub
![1702954573296](https://github.com/Azure/azure-sdk-for-net/assets/53285941/6d125294-f4c7-4204-8071-7be1074047a7)

Please let me known if furthure clarification is required. 

Such issue is very important to us. You see, yesterday (Monday) when I came back office and just did one normal routine chceking,  and foud out it was stopped two days ago at Satursday night. And then I have to kill the pod manually, from log it looks like zombies as no more log generated out, but my k8s health probe still working (which is based upon linux shell command to compare two files' content, both store number of the current epoch mills, one file is kept being updated by my long-running thread, and after comparing, this file conent would be copied into another file.)  As k8s health probe does't kill my pod, it means my pods still working. So, I need improve health probe logic. Anyway, health probe is another story, I need make such wierd thing fixed (all 5 eventhub receivers share one same UUID).

**Comment by anlong465** [comment]
BTW, because of the column length limitation, the exception stack trace was truncated, and the below here is one similar snapshot
![1702966140005](https://github.com/Azure/azure-sdk-for-net/assets/53285941/dc152d90-e5a2-4611-b04f-496d2fb13f6e)

**Comment by jsquire** [comment]
Hi @anlong465.  Thank you for the context.   Since you're using Java, I'm going to transfer this to the Azure SDK for Java repository, where the team members best able to assist will see it
